Edge case. When cropmodel is used, and the detection model returns no result, the assertion is raised.

```
def test_predict_tile_with_crop_model_empty():
    """If the model return is empty, the crop model should return an empty dataframe"""
    raster_path = get_data("SOAP_061.png")
    m = main.deepforest()
    patch_size = 400
    patch_overlap = 0.05
    iou_threshold = 0.15
    mosaic = True
    # Set up the crop model
    crop_model = model.CropModel()

    # Call the predict_tile method with the crop_model
    m.config["train"]["fast_dev_run"] = False
    m.create_trainer()
    result = m.predict_tile(raster_path=raster_path,
                            patch_size=patch_size,
                            patch_overlap=patch_overlap,
                            iou_threshold=iou_threshold,
                            mosaic=mosaic,
                            crop_model=crop_model)

    # Assert the result
    assert isinstance(result, pd.DataFrame)
    assert set(result.columns) == {
        "xmin", "ymin", "xmax", "ymax", "label", "score", "cropmodel_label", "geometry",
        "cropmodel_score", "image_path"
    }
```

```
self = <deepforest.dataset.BoundingBoxDataset object at 0x1849afdd0>
df = Empty DataFrame
Columns: [xmin, ymin, xmax, ymax, label, score, image_path]
Index: []
root_dir = '/Users/benweinstein/Documents/DeepForest/src/deepforest/data'
transform = None, augment = False

    def __init__(self, df, root_dir, transform=None, augment=False):
        self.df = df
    
        if transform is None:
            self.transform = bounding_box_transform(augment=augment)
        else:
            self.transform = transform
    
        unique_image = self.df['image_path'].unique()
>       assert len(unique_image
                  ) == 1, "There should be only one unique image for this class object"
E       AssertionError: There should be only one unique image for this class object
```

I am assigning to myself and will fix shortly. I found it while working with the CropModel workflow. Just need to check for results.empty.

--- a/src/deepforest/predict.py
+++ b/src/deepforest/predict.py
@@ -212,6 +212,10 @@ def _predict_crop_model_(crop_model,
     Returns:
         The updated results dataframe with predicted labels and scores.
     """
+    if results.empty:
+        print("No predictions to run crop model on, returning empty dataframe")
+        return results
+
     bounding_box_dataset = dataset.BoundingBoxDataset(
         results,
         root_dir=os.path.dirname(raster_path),

--- a/tests/test_main.py
+++ b/tests/test_main.py
@@ -231,7 +231,7 @@ def test_predict_image_fromarray(m):
         prediction = m.predict_image(image=image)
 
     assert isinstance(prediction, pd.DataFrame)
-    assert set(prediction.columns) == {"xmin", "ymin", "xmax", "ymax", "label", "score", "geometry"} 
+    assert set(prediction.columns) == {"xmin", "ymin", "xmax", "ymax", "label", "score", "geometry"}
     assert not hasattr(prediction, 'root_dir')
 
 def test_predict_big_file(m, tmpdir):
@@ -649,3 +649,28 @@ def test_predict_tile_with_crop_model(m, config):
         "xmin", "ymin", "xmax", "ymax", "label", "score", "cropmodel_label", "geometry",
         "cropmodel_score", "image_path"
     }
+
+
+def test_predict_tile_with_crop_model_empty():
+    """If the model return is empty, the crop model should return an empty dataframe"""
+    raster_path = get_data("SOAP_061.png")
+    m = main.deepforest()
+    patch_size = 400
+    patch_overlap = 0.05
+    iou_threshold = 0.15
+    mosaic = True
+    # Set up the crop model
+    crop_model = model.CropModel()
+
+    # Call the predict_tile method with the crop_model
+    m.config["train"]["fast_dev_run"] = False
+    m.create_trainer()
+    result = m.predict_tile(raster_path=raster_path,
+                            patch_size=patch_size,
+                            patch_overlap=patch_overlap,
+                            iou_threshold=iou_threshold,
+                            mosaic=mosaic,
+                            crop_model=crop_model)
+
+    # Assert the result
+    assert result is None

